### PR TITLE
Feature/ins 184 contacts tableview

### PR DESCRIPTION
**Pull request by vierrra**

### CODE REVIEW

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 1]
```diff
@@ -18,7 +19,19 @@ public class ContactsCard: UITableViewCell {
     public var phoneButton:       UIButton?
     public var presenterDelegate: ContactsTableViewShowPresenter?
     public var photoImageView:    RoundedImageView?
-       
+    var person: ContactsInformation?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Indentation

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 2]
```diff
@@ -18,7 +19,19 @@ public class ContactsCard: UITableViewCell {
...
+    
+    var callToMessage = String()
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Indent this block (lines 24 - 31)

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 3]
```diff
@@ -51,6 +64,8 @@ public class ContactsCard: UITableViewCell {
     }
     
     private func setButtonsActionsFor(mail: String, phone: String) {
+        person = ContactsInformation(name: "", role: "", photo: UIImage(), mail: mail, phone: phone)
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Break the parameters and indent them.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 4]
```diff
@@ -60,30 +75,92 @@ public class ContactsCard: UITableViewCell {
...
+        alert.addAction(UIAlertAction(title: self.makeCall + "        ", style: .default, handler: { (_) in
+            let url = URL(string: "TEL://\(currentPerson.phone as! String)")
+            if let url = url {
+                UIApplication.shared.open(url, options: [:], completionHandler: nil)
+           }
+        }))
+               
+        alert.addAction(UIAlertAction(title: self.cancel, style: .cancel, handler: { (_) in
+           }))
+
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Remove extra line.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 5]
```diff
@@ -60,30 +75,92 @@ public class ContactsCard: UITableViewCell {
...
     
+    
+    func showAlertToGoGuidance() {
+        let alert = UIAlertController(title: self.supportWarningMessage, message: self.warningMessage, preferredStyle: .alert)
+                
+        alert.addAction(UIAlertAction(title: self.guidance + "       ", style: .default, handler: { (_) in
+                if let url = URL(string: "https://support.apple.com/pt-br/HT201320") {
+                    UIApplication.shared.open(url)
+                }
+            }))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Indent it with the alert.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 6]
```diff
@@ -60,30 +75,92 @@ public class ContactsCard: UITableViewCell {
...
+                
+        alert.addAction(UIAlertAction(title: self.cancel, style: .cancel, handler: { (_) in
+            }))
+            
+            if let present = presenterDelegate {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: This block isn't inside the alert handler, so indent it back.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 7]
```diff
@@ -46,7 +54,14 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
         
         self.configureEmptyMessageLabel()
         self.configureSearchView()
-        self.configureTableView()
+        self.configureTableView(callToMessage: callToMessage,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Indent this block.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 8]
```diff
@@ -64,7 +79,7 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
                                      placeholderText: searchBarPlaceholder)
     }
     
-    fileprivate func configureTableView() {
+    fileprivate func configureTableView(callToMessage: String, mailMessage: String, supportWarningMessage: String, makeCall: String, sendMail: String, warningMessage: String, guidance: String, cancel: String){
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 9]
```diff
@@ -74,10 +89,19 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
         
         tableView?.configureTableViewWith(contactsList:   contactsSiteList,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Indent this block.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 10]
```diff
@@ -21,6 +21,14 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
     public var selectCellDelegate: ContactsTableViewDidSelectDelegate?
     
     public var presenterDelegate:  UIViewController?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift:23]: Indent this block.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 11]
```diff
@@ -32,7 +40,15 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
     
     public func configureTableViewWith(contactsList:   [ContactsInformation],
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 12]
```diff
@@ -41,6 +57,14 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
         self.selectCellDelegate   = selectCellDelegate
         self.filteredContactsList = contactsList
         self.allContactsList      = contactsList
+        self.callToMessage = callToMessage
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/CellWithImage/ProfileCellTableViewController.swift [hunk 13]
```diff
@@ -61,9 +61,9 @@ class ProfileCellTableViewController: UITableViewController {
         self.setButtonsActionsFor(index: indexPath.row)
         
         cell?.set(mailButton: self.validMailButtons[indexPath.row],
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/CellWithImage/ProfileCellTableViewController.swift:63]: Indent this parameters.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 14]
```diff
@@ -7,18 +7,25 @@
...
-    let contactPhoto                 = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT0P8RsPCO0qj21UeIfaVkRcsqguonI6bP4iLr3tWwS4qIS4MSquw"
+    let contactPhoto                 = UIImage(named: "godfatherWashington") ?? UIImage()
     let contactEmail                 = "stant@stant.com.br"
-    let contactPhone                 = "99999-9999"
+    let contactPhone                 = "82999999999"
     
     var contactsSiteList = [ContactsInformation]()
     
+    @IBAction func alertInformation() {
+        let alert = UIAlertController(title: "Warning", message: "select a construction site to load the work contacts", preferredStyle: .alert)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Break and indent the paremeters.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 15]
```diff
@@ -30,39 +37,56 @@ class ConstructionContactsViewController: UIViewController, ContactsTableViewSho
...
-                                 bottom:   0,
-                                 right:    0))
-        
-        contactsTableView.configureViewWith(contactsList: contactsSiteList,
-                                            searchBarIcon: UIImage(named: "search") ?? UIImage(),
-        searchBarPlaceholder: "Search",
-        tableViewDelegate: self,
-        emptyMessage: "No construction site to show on current screen. Please try again later.")
+        if !contactsSiteList.isEmpty {
+            let contactsTableView = ContactsTableViewWithCollapseSearchBar(frame: CGRect(x: 0,
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 1 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 2 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 3 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 4 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 5 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 6 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 7 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 8 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 9 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 10 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift:23]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 11 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 12 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/CellWithImage/ProfileCellTableViewController.swift (hunk 13 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/CellWithImage/ProfileCellTableViewController.swift:63]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 14 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 15 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

**Review by Leandro97 — APPROVED** [review]

**Review by andersonfernandes — APPROVED** [review]

**Review by Leandro97 — APPROVED** [review]

**Pull request merged**